### ISSUE TITLE
When cc rules depend on objc_library, headers of transitive dependencies are dropped 

### ISSUE THREAD

**Issue by cpsauer** [bazelbuild/bazel#10738]
Hi wonderful Bazel folk,

I'm pretty sure I've found a bug in the interaction between `cc` rules and `objc_library`, which makes them hard to use together in meaningfully sized dependency graphs. 

Offhand, my guess is that the fix would need to be on the `cc` side.

### Background:
C-style language rules (`cc_library`, `objc_library`, etc.) need to propagate their `hrs` to all rules that might transitively `#include` them. In practice, that means that the `hrs` need to be propagated to *all* c-style rules that depend on them, transitively or directly. In Bazel, this already works when `cc` rules depend on each other, when `objc` rules depend on each other, and when `objc` rules depend on `cc` rules.

### The Issue
The case that's broken is `cc` rules depending on `objc` rules. The `hdrs` of the `objc` rules' dependencies get dropped, and the build fails when the `cc` rule tries to `#include` on the the headers of an `objc` rule that `#includes` its dependencies.

#### Here's a minimal example of the issue. 
[Test.zip](https://github.com/bazelbuild/bazel/files/4178037/Test.zip)
- It contains three targets: A, B, and C.
- A.c `#include`s B.h, which `#include`s C.h
    - Therefore, the include graph looks like A -> B -> C.
    - The deps graph is hooked up the same way, so there should never be missing headers

Now we'll try building `//:A`: `bazel build A`:
If A, B, and C are all `cc_library`s, this works. Same if they're all `objc_libraries`.
All the following cases work as expected:
- cc -> cc -> cc
- cc -> cc -> objc
- objc -> objc -> objc
- objc -> objc -> cc
- objc -> cc -> objc
- objc -> cc -> cc

These two cases are broken (project left configured in the first of these cases):
- cc -> objc -> cc
- cc -> objc -> objc

In both cases, building fails with the error
```
 In file included from A.c:1:
./B.h:1:10: fatal error: 'C.h' file not found
#include "C.h"
```

#### My conclusion
`cc` rules seem to fail to pick up the `hdrs` of the `deps` of `objc` rules. (Other cases work as expected.) This happens whether those dependencies are `cc` rules or `objc` rules. The expected behavior would be that `objc` rules would propagate the `hdrs` of their `deps` the same way as `cc` rules do, or `objc` rules do with each other.

#### Severity
You can work around this issue, but it's fairly ugly. This issue would come up whenever someone uses an objc_library to bridge from C/C++ into a meaningfully-sized Objective-C or Objective-C++ implementation, as happens with cross-platform code with a backend for Apple platforms.

Thanks,
Chris
(ex-Googler)

**Comment by cpsauer** [comment]
It would appear that there's a related issue where archives get dropped when cc_library -> objc_library -> objc_import. Discovered that while working around #10740

**Comment by cpsauer** [comment]
This was fixed, I'm pretty sure by #10674--given what I've learned about Bazel since writing this long ago.

Thanks, @googlewalt! Really appreciate your making C++ -> Obj-C dependencies work :)
